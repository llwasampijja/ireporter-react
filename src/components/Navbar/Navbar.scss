
/* header */
header {
  /* background-color: rgb(65, 105, 225); */
  background-color: rgb(65, 105, 225);
  padding: 15px;
  color: #FFFFFF;
}

header.navigation {
  border-bottom: #E8E8E8 solid 1px;
}

header h1 {
  display: inline-block;
  margin: auto 0px;
}

header h1 a:link, header h1 a:visited{
  color: #FFFFFF;
}

header h1 a:hover{
  color: rgb(65, 105, 225);
}

header nav ul li {
  display: inline-block;
  font-weight: bold;
}

header nav ul li a {
  padding: 10px 10px 10px 10px;
}

header nav ul li a:link{
  color: #FFFFFF;
}

header nav ul li a:hover{
  color: rgb(65, 105, 225);
}

/* dashboard styles */
.bigscreen-menu-nav {
  display: inline-block;
  float: right;
}

.bigscreen-menu-nav ul {
  height: 50px;
}
.header-nav-item {
  margin: auto;
}

.header-nav-item a:visited{
  color: rgb(250, 246, 246);
}