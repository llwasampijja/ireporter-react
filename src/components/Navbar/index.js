// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// styles
import './Navbar.scss';

const Navbar = (props) => {
  const {
    userName,
    redflagsLink,
    adminPanel,
  } = props;
  return (
    <header className="navigation">
      <h1><a href="index.html">IReporter</a></h1>
      <nav className="bigscreen-menu-nav" id="bigscreen-menu-nav-signedin">
        <ul>
          <li className="header-nav-item"><a href={adminPanel}>{redflagsLink}</a></li>
          <li className="header-nav-item non-admin-link"><a href="/sign-up">{userName}</a></li>
        </ul>
      </nav>
    </header>
  );
};

Navbar.propTypes = {
  userName: PropTypes.string,
  redflagsLink: PropTypes.string,
  adminPanel: PropTypes.string,
};

Navbar.defaultProps = {
  userName: '',
  redflagsLink: '',
  adminPanel: '',
};

export default Navbar;
