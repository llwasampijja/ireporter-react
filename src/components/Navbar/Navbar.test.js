// react libraries
import React from 'react';

// third-[arty libraries
import { shallow } from 'enzyme';

// components
import Navbar from '.';

describe('tests for the Navbar component', () => {
  it('Navbar renders properly', () => {
    const component = shallow(<Navbar />);
    expect(component).toMatchSnapshot();
  });
});
