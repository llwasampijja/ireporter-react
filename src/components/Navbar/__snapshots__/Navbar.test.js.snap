// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the Navbar component Navbar renders properly 1`] = `
<header
  className="navigation"
>
  <h1>
    <a
      href="index.html"
    >
      IReporter
    </a>
  </h1>
  <nav
    className="bigscreen-menu-nav"
    id="bigscreen-menu-nav-signedin"
  >
    <ul>
      <li
        className="header-nav-item"
      >
        <a
          href=""
        />
      </li>
      <li
        className="header-nav-item non-admin-link"
      >
        <a
          href="/sign-up"
        />
      </li>
    </ul>
  </nav>
</header>
`;
